omod-common/src/test/java/org/openmrs/module/webservices/rest/web/api/impl/RestServiceImplTest.java

```diff
@@ -268,4 +271,161 @@ public void getResourceBySupportedClass_shouldReturnTheMostExactMatch() throws E
+		assertFalse(compositeKey2.equals(compositeKey1));
+	}
+	
+	/**
+	 * @verifies return false if given a composite key with different additional key
+	 * @see RestServiceImpl.CompositeSearchHandlerKey#equals(Object)
+	 */
+	@Test
+	public void equals_shouldReturnFalseIfGivenACompositeKeyWithDifferentAdditionalKey() throws Exception {
+		Class compositeSearchHandlerClass = Class
+		        .forName("org.openmrs.module.webservices.rest.web.api.impl.RestServiceImpl$CompositeSearchHandlerKey");
+		Constructor ctor = compositeSearchHandlerClass.getDeclaredConstructors()[0];
+		
+		Object compositeKey1 = ctor.newInstance("v1/order", "default");
+		Object compositeKey2 = ctor.newInstance("v2/order", "custom");
+		assertFalse(compositeKey1.equals(compositeKey2));
+		assertFalse(compositeKey2.equals(compositeKey1));
+	}
+	
+	/**
+	 * @verifies return false if given an object which is not an instanceof this class
+	 * @see RestServiceImpl.CompositeSearchHandlerKey#equals(Object)
+	 */
+	@Test
+	public void equals_shouldReturnFalseIfGivenAnObjectWhichIsNotAnInstanceofThisClass() throws Exception {
+		Class compositeSearchHandlerClass = Class
+		        .forName("org.openmrs.module.webservices.rest.web.api.impl.RestServiceImpl$CompositeSearchHandlerKey");
+		Constructor ctor = compositeSearchHandlerClass.getDeclaredConstructors()[0];
+		
+		Object compositeKey1 = ctor.newInstance("v1/order", "default");
+		assertFalse(compositeKey1.equals("otherClass"));
+	}
+	
+	/**
+	 * @verifies return false if given a composite key with null as supported resource
+	 * @see RestServiceImpl.CompositeSearchHandlerKey#equals(Object)
+	 */
+	@Test
+	public void equals_shouldReturnFalseIfGivenACompositeKeyWithNullAsSupportedResource() throws Exception {
+		Class compositeSearchHandlerClass = Class
+		        .forName("org.openmrs.module.webservices.rest.web.api.impl.RestServiceImpl$CompositeSearchHandlerKey");
+		Constructor ctor = compositeSearchHandlerClass.getDeclaredConstructors()[0];
+		
+		Object compositeKey1 = ctor.newInstance("v1/order", "default");
+		Object compositeKey2 = ctor.newInstance("v2/order", null);
+		assertFalse(compositeKey1.equals(compositeKey2));
+		assertFalse(compositeKey2.equals(compositeKey1));
+	}
+	
+	/**
+	 * @verifies return false if given a composite key with null as additional key
+	 * @see RestServiceImpl.CompositeSearchHandlerKey#equals(Object)
+	 */
+	@Test
+	public void equals_shouldReturnFalseIfGivenACompositeKeyWithNullAsAdditionalKey() throws Exception {
+		Class compositeSearchHandlerClass = Class
+		        .forName("org.openmrs.module.webservices.rest.web.api.impl.RestServiceImpl$CompositeSearchHandlerKey");
+		Constructor ctor = compositeSearchHandlerClass.getDeclaredConstructors()[0];
+		
+		Object compositeKey1 = ctor.newInstance("v1/order", "default");
+		Object compositeKey2 = ctor.newInstance(null, "default");
+		assertFalse(compositeKey1.equals(compositeKey2));
+		assertFalse(compositeKey2.equals(compositeKey1));
```

Comment:
Yes, it can never happen, because it is prevented much earlier when using public api we expose. The test is still useful though. Simply expect IlegalArgumentException.class when passing null as supportedResource to the SearchConfig constructor.